fe/lib/components/LocationSelectMap/LocationSelectMap.tsx

```diff
@@ -0,0 +1,191 @@
+import { ApolloClient, useQuery } from '@apollo/client';
+import {
+  Alert,
+  Box,
+  ButtonIcon,
+  Column,
+  Columns,
+  IconClear,
+  Stack,
+  Text,
+  useColor,
+  useResponsiveValue,
+} from 'braid-design-system';
+import { Map, Marker, Overlay, ZoomControl } from 'pigeon-maps';
+import React from 'react';
+import { useState } from 'react';
+import { InlineCode } from 'scoobie';
+import { useDebounce } from 'use-debounce';
+
+import { Location, NearbyLocationsQuery } from '../../types/seekApi.graphql';
+import { formatPoint } from '../../utils/formatPoint';
+
+import { NEAREST_LOCATIONS } from './queries';
+
+import * as styles from './styles.css';
+
+const DEFAULT_ZOOM = 10;
+const DEFAULT_MAP_HEIGHT = 400;
+
+export interface LocationSelectMapProps {
+  debounceDelay?: number;
+  schemeId: string;
+  onLocationSelected: (location: Location) => void;
+  initialLocation: [number, number];
+  client?: ApolloClient<unknown>;
+  mapHeight?: {
+    wide: number;
+    desktop: number;
+    tablet: number;
+    mobile: number;
+  };
+}
+
+export const LocationSelectMap = ({
+  debounceDelay = 250,
+  schemeId,
+  onLocationSelected,
+  initialLocation,
+  client,
+  mapHeight = {
+    wide: DEFAULT_MAP_HEIGHT,
+    desktop: DEFAULT_MAP_HEIGHT,
+    tablet: 200,
+    mobile: 200,
+  },
+}: LocationSelectMapProps) => {
+  const [latLong, setLatLong] = useState(initialLocation);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const responsiveValue = useResponsiveValue();
+  const color = useColor();
+  const [debouncedLatLong] = useDebounce(latLong, debounceDelay);
+
+  const { data, previousData } = useQuery<NearbyLocationsQuery>(
+    NEAREST_LOCATIONS,
+    {
+      variables: {
+        geoLocation: {
+          latitude: debouncedLatLong[0],
+          longitude: debouncedLatLong[1],
+        },
+        schemeId,
+      },
+      client,
+    },
+  );
+
+  const nearestLocations =
+    data?.nearestLocations ?? previousData?.nearestLocations;
+
+  const isDesktopOrAbove = responsiveValue({
+    mobile: false,
+    desktop: true,
+  });
```

Comment:
While this isn't my first preference because it imposes runtime overhead, I think the layout you're going for would be a bit difficult to accomplish in pure CSS 👍